weave/stitch.py

```diff
@@ -307,6 +328,30 @@ def stitch_node_inner(
         inputs[0].tags["groupKey"] = groupkey
         # And we return the original object
         return inputs[0]
+    elif node.from_op.name.endswith("join"):
+        fn1 = inputs[2].val
+        fn2 = inputs[3].val
+        if fn1 is None:
+            raise errors.WeaveInternalError("non-const not yet supported")
+        if fn2 is None:
+            raise errors.WeaveInternalError("non-const not yet supported")
+        alias1 = inputs[4].val
+        alias2 = inputs[5].val
+        if alias1 is None:
+            raise errors.WeaveInternalError("non-const not yet supported")
+        if alias2 is None:
+            raise errors.WeaveInternalError("non-const not yet supported")
+        joinKey1 = subgraph_stitch(fn1, {"row": inputs[0]}, sg)
+        joinKey2 = subgraph_stitch(fn2, {"row": inputs[1]}, sg)
+        result = LiteralDictObjectRecorder(
+            node, val={alias1: inputs[0], alias2: inputs[1]}
+        )
+        # We only pass joinKey1 in as tag. I think this is ok, as stitch has already determined
+        # that everything in the tag is a necessary column (we needed those columns to produce
+        # the join key in the first place).
+        # But I'm not totally sure!
+        result.tags["joinObj"] = joinKey1
```

Comment:
I think we could do something like `opCoalesce(joinKey1, joinKey2)`... need to write tests to be sure